test/e2e/specs/editor/plugins/templates.spec.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'templates', () => {
+	test.describe( 'Using a CPT with a predefined template', () => {
+		test.beforeAll( async ( { requestUtils } ) => {
+			await requestUtils.activatePlugin(
+				'gutenberg-test-plugin-templates'
+			);
+		} );
+
+		test.beforeEach( async ( { admin } ) => {
+			await admin.createNewPost( { postType: 'book' } );
+		} );
+
+		test.afterAll( async ( { requestUtils } ) => {
+			await requestUtils.deactivatePlugin(
+				'gutenberg-test-plugin-templates'
+			);
+		} );
+
+		test( 'Should add a custom post types with a predefined template', async ( {
+			editor,
+		} ) => {
+			expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		test( 'Should respect user edits to not re-apply template after save (single block removal)', async ( {
+			page,
+			editor,
+		} ) => {
+			const beforeContent = await editor.getEditedPostContent();
+
+			// Remove a block from the template to verify that it's not
+			// re-added after saving and reloading the editor.
+			await page.focus( 'role=textbox[name="Add title"i]' );
+			await page.keyboard.press( 'ArrowDown' );
+			await page.keyboard.press( 'Backspace' );
+			await page.click( 'role=button[name="Save draft"i]' );
+			await expect(
+				page.locator(
+					'role=button[name="Dismiss this notice"i] >> text=Draft saved'
+				)
+			).toBeVisible();
+			await page.reload();
+
+			const expectedContent = await page.evaluate( ( content ) => {
+				const blocks = window.wp.blocks.parse( content );
+				blocks.splice( 0, 1 );
+				return window.wp.blocks.serialize( blocks );
+			}, beforeContent );
+
+			await expect
+				.poll( editor.getEditedPostContent )
+				.toBe( expectedContent );
```

Comment:
This looks like it could be an assertion against the snapshot. What do you think?

Reply:
I think it depends on the intention of the test. If the test wants to test the full output of the result then we can use snapshot testing. However, if it's any other intent, then testing against that intent would be better IMO. In this case, we're testing if deleting the first image block would reflect on the saved template, so I think we should test if the result has that first image block. Furthermore, we already have snapshot testing in the first test case, so that's already covered.

Reply:
That makes sense. Thanks for clarifying.